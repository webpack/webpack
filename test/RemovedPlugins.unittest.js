const webpack = require("../lib/webpack");
const RemovedPluginError = require("../lib/RemovedPluginError");

describe("removed plugin errors", () => {
	it("should error when accessing removed plugins", () => {
		expect(() => webpack.optimize.UglifyJsPlugin).toThrow(RemovedPluginError);
		expect(
			() => webpack.optimize.UglifyJsPlugin
		).toThrowErrorMatchingSnapshot();

		expect(() => webpack.optimize.CommonsChunkPlugin).toThrow(
			RemovedPluginError
		);
		expect(
			() => webpack.optimize.CommonsChunkPlugin
		).toThrowErrorMatchingSnapshot();
	});
});
