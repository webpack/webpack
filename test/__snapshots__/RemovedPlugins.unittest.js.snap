// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removed plugin errors should error when accessing removed plugins 1`] = `"webpack.optimize.UglifyJsPlugin has been removed, please use config.optimization.minimize instead."`;

exports[`removed plugin errors should error when accessing removed plugins 2`] = `"webpack.optimize.CommonsChunkPlugin has been removed, please use config.optimization.splitChunks instead."`;
